src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1370,6 +1371,15 @@ BoundExpression bindDefaultArgument(SyntaxNode syntax, ParameterSymbol parameter
                     var memberName = containingMember.GetMemberCallerName();
                     defaultValue = new BoundLiteral(syntax, ConstantValue.Create(memberName), Compilation.GetSpecialType(SpecialType.System_String)) { WasCompilerGenerated = true };
                 }
+                else if (callerSourceLocation is object && getArgumentIndex(parameter.CallerArgumentExpressionParameterIndex, argsToParamsOpt) is int argumentIndex &&
+                    argumentIndex > -1 && argumentIndex < arguments.Length)
```

Comment:
>argumentIndex < arguments.Length [](start = 42, length = 32)

Can this condition ever be false?

Reply:
@AlekseyTs Yes it can happen when the caller argument expression is referring to an optional parameter that's not passed.